src/parcsr_ls/schwarz.c

```diff
@@ -3312,8 +3359,7 @@ hypre_ParAMGCreateDomainDof(hypre_ParCSRMatrix   *A,
    }
    else
    {
-      nf = 1;
-      if (domain_type == 1) { nf = num_functions; }
+      nf = (domain_type == 1) ? num_functions : 1;
```

Comment:
In this case, the compiler shouldn't have complained.  Out of curiosity, if you write this as an `if/else` block, will it generate a warning?  The `?:` syntax isn't my favorite, though it is short, concise, and should be clear to the compiler that the variable is being initialized.

Reply:
In fact, the previous form didn't lead to a warning. I guess I made this change because I like the ternary operator and I was using it in a couple of places nearby :)

I can revert the change also...

Reply:
No need.  I was just curious.